react-components/src/apps/admin/AppStats.js

```diff
@@ -0,0 +1,328 @@
+import ToolbarSeparator from "material-ui-next/Toolbar";
+import TextField from "material-ui-next/TextField";
+import Button from "material-ui-next/Button";
+import {withStyles} from "material-ui-next/styles";
+import Table, {
+    TableHead,
+    TableBody,
+    TableCell,
+    TableFooter,
+    TablePagination,
+    TableRow
+} from "material-ui-next/Table";
+import IconButton from "material-ui-next/IconButton";
+import FirstPageIcon from "@material-ui/icons/FirstPage";
+import KeyboardArrowLeft from "@material-ui/icons/KeyboardArrowLeft";
+import KeyboardArrowRight from "@material-ui/icons/KeyboardArrowRight";
+import LastPageIcon from "@material-ui/icons/LastPage";
+import PropTypes from "prop-types";
+import moment  from "moment";
+import RefreshIndicator from "material-ui/RefreshIndicator";
+
+
+const pagingStyles = theme => ({
+    root: {
+        flexShrink: 0,
+        color: theme.palette.text.secondary,
+        marginLeft: theme.spacing.unit * 2.5,
+
+    },
+});
+
+class TablePaginationActions extends React.Component {
+    handleFirstPageButtonClick = event => {
+        this.props.onChangePage(event, 0);
+    };
+
+    handleBackButtonClick = event => {
+        this.props.onChangePage(event, this.props.page - 1);
+    };
+
+    handleNextButtonClick = event => {
+        this.props.onChangePage(event, this.props.page + 1);
+    };
+
+    handleLastPageButtonClick = event => {
+        this.props.onChangePage(
+            event,
+            Math.max(0, Math.ceil(this.props.count / this.props.rowsPerPage) - 1),
+        );
+    };
+
+    render() {
+        const {classes, count, page, rowsPerPage, theme, appearance} = this.props;
+
+        return (
+            <div className={classes.root}>
+                <IconButton
+                    onClick={this.handleFirstPageButtonClick}
+                    disabled={page === 0}
+                    aria-label='First Page'
+                >
+                    {theme.direction === 'rtl' ? <LastPageIcon /> : <FirstPageIcon />}
+                </IconButton>
+                <IconButton
+                    onClick={this.handleBackButtonClick}
+                    disabled={page === 0}
+                    aria-label='Previous Page'
+                >
+                    {theme.direction === 'rtl' ? <KeyboardArrowRight /> : <KeyboardArrowLeft />}
+                </IconButton>
+                <IconButton
+                    onClick={this.handleNextButtonClick}
+                    disabled={page >= Math.ceil(count / rowsPerPage) - 1}
+                    aria-label='Next Page'
+                >
+                    {theme.direction === 'rtl' ? <KeyboardArrowLeft /> : <KeyboardArrowRight />}
+                </IconButton>
+                <IconButton
+                    onClick={this.handleLastPageButtonClick}
+                    disabled={page >= Math.ceil(count / rowsPerPage) - 1}
+                    aria-label='Last Page'
```

Comment:
Should these First/Previous/Next/Last Page labels be moved to the appearance?  There are also a couple different styles defined that maybe should move as well?

Reply:
I wanted to but I couldn't figure out how to pass that in `props` yet!